ebill_postfinance/models/account_move.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+import logging
+
+import odoo
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.pdf import merge_pdf
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.onchange("transmit_method_id")
+    def _onchange_transmit_method(self):
+        if self.move_type not in ("out_invoice", "out_refund"):
+            return
+        postfinance_method = self.env.ref(
+            "ebill_postfinance.postfinance_transmit_method"
+        )
+        if self.transmit_method_id == postfinance_method:
+            contract = self.partner_id.get_active_contract(self.transmit_method_id)
+            if contract:
+                self.partner_bank_id = contract.postfinance_service_id.partner_bank_id
```

Comment:
I think this should be done in `_compute_partner_bank_id` instead, and get rid of `_onchange_transmit_method`